### DISCUSSION TITLE
Extensions for DGD 1.7.1

### DISCUSSION THREAD

**Discussion by dworkin** [dworkin/dgd#84] [General]
These extension modules are available:
* JIT compiler module. This is probably the only one that's relevant when DGD is used to run a MUD. Can be compiled on Windows. Stable and recommended for everyone.
* Zlib compression module. Stable. Does not currently compile on Windows.
* Crypto module. Frequently updated, but usable. Does not currently compile on Windows.
* TLS module. Stable, but not usable for Hydra, or if you use hotbooting in DGD. Does not currently compile on Windows.

**Comment by dworkin** [comment]
There is a new extension, [libsignal](https://github.com/LPC-language/libsignal), which has its own repository. It is needed to run [signal-server](https://github.com/LPC-language/signal-server).

**Comment by geirpg** [comment]
I might be blind, but I can't find the extension modules - spesifically Crypto..
What I'm really looking for, is to see if Argon2id will be available as a password hashing algoritm under DGD, for use with VikingMud.

**Comment by dworkin** [comment]
Extension module framework: https://github.com/dworkin/lpc-ext
Crypto extension: https://github.com/dworkin/lpc-ext/blob/master/src/kfun/crypto.c

**Comment by dworkin** [comment]
Compiled extension modules for linux i386 (32 bits), suitable for Hydra: https://download.dworkin.nl/hydra/ext
